### DISCUSSION TITLE
How to redirect to https

### DISCUSSION THREAD

**Discussion by zhuravkov** [salvo-rs/salvo#383] [General]
Hello, I can't figure out how to redirect to https correctly! it turned out just like that, but it’s not quite right, because you can change the scheme, but it didn’t work out for me

`#[handler]
async fn redir(req: &mut Request) -> Redirect{
    let a = req.scheme_mut();
    *a = Scheme::HTTPS;
    println!("{}",req.scheme());  //print "https"
    println!("{}",req.uri());           // print "http://domain.com/"
    let c: String = req.uri().to_string().replace("http", "https"); // This work, but its not good
    Redirect::other(c)
}

#[tokio::main]
async fn main() {
    tracing_subscriber::fmt().init();

    let mut router = Router::new()
        .push(Router::new().scheme(Scheme::HTTP).handle(redir))
        .push(Router::new().get(hello))`

**Comment by chrislearn** [comment]
Maybe you can use `ForceHttps`.

https://github.com/salvo-rs/salvo/blob/main/crates/extra/src/force_https.rs

**Comment by zhuravkov** [comment]
Yes it works! Thank you! You have developed an excellent framework, it is as convenient and concise as possible, but the only source of information on working with it is its documentation, which is not yet completely complete, so a number of questions arise in solving typical tasks when mastering it, thank you for answering questions right away! I believe in the success of Salvo!

```
use salvo::prelude::*;
use salvo::logging::Logger;


#[handler]
async fn hello() -> &'static str {
    "Hello World"
}

#[handler]
async fn data(req: &mut Request, res: &mut Response)  {
    let param = req.params().get("param").unwrap();
    let content = format!("With param: {}",param);
    res.render(Text::Html(content));
}

#[tokio::main]
async fn main() {
    tracing_subscriber::fmt().init();
    let mut router = Router::new()
        .hoop(ForceHttps::new().https_port(443))
        .hoop(Logger::new())
        .push(Router::new().get(hello))
        .push(Router::with_path("<param>").get(data))
    ;
    let listener: AcmeListener<TcpListener<&str>> = TcpListener::new("0.0.0.0:443")
        .acme()
        .cache_path("acme/letsencrypt")
        .add_domain("mydomain.com")
        .http01_challege(&mut router);
    let acceptor = listener.join(TcpListener::new("0.0.0.0:80")).bind().await;
    Server::new(acceptor).serve(router).await;
}

```
